pkg/transaction/transaction_test.go

```diff
@@ -634,13 +662,17 @@ func TestTransactionCancel(t *testing.T) {
 	t.Run("ok", func(t *testing.T) {
 		t.Parallel()
 
-		cancelTx := types.NewTx(&types.LegacyTx{
-			Nonce:    nonce,
-			To:       &recipient,
-			Value:    big.NewInt(0),
-			Gas:      21000,
-			GasPrice: new(big.Int).Add(gasPrice, big.NewInt(1)),
-			Data:     []byte{},
+		price := new(big.Int).Add(gasPrice, big.NewInt(1))
+
+		cancelTx := types.NewTx(&types.DynamicFeeTx{
+			ChainID:   chainID,
+			Nonce:     nonce,
+			To:        &recipient,
+			Value:     big.NewInt(0),
+			Gas:       21000,
+			GasTipCap: price,
+			GasFeeCap: price,
```

Comment:
It wasn't the last, and neither is this, but I'll leave it to you to apply the suggestion to all tests or none.